src/core/util/options.js

```diff
@@ -379,10 +379,10 @@ export function mergeOptions (
   normalizeInject(child, vm)
   normalizeDirectives(child)
   const extendsFrom = child.extends
-  if (extendsFrom) {
+  if (extendsFrom && !child._base) {
```

Comment:
Not sure if there's some other flag that's supposed to be used for this, but the presence of `_base` seemed like a decent indicator at a glance.